src/main/java/com/pscheduler/server/repository/CourseGPARepository.java

```diff
@@ -1,8 +1,53 @@
 package com.pscheduler.server.repository;
 
 import com.pscheduler.server.model.CourseGPA;
-import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.data.repository.query.Param;
 import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+import org.springframework.data.rest.core.annotation.RestResource;
 
 @RepositoryRestResource(collectionResourceRel="gpa", path="gpa")
-public interface CourseGPARepository extends CrudRepository<CourseGPA, Integer> {}
+public interface CourseGPARepository extends PagingAndSortingRepository<CourseGPA, Integer> {
+    @Query(
+        "SELECT course FROM CourseGPA course"
```

Comment:
Let's think about this query. I don't think it would really be useful to query the exact percentage of a grade or the GPA, or even the number of students and withdraws.

Here are some ideas:
- `min`/`max` parameters to search for GPA
- `>=`/`<=` parameter for each grade though this one probably adds too many parameters
  - Or maybe just 1 of them, you can probably find any kind of distributions of grades if you set them correctly
- Remove student/withdraws I don't see a use case for this

Thoughts?

Reply:
The 'min/max' parameters sounds like a cool idea, but I'm not sure if it'll have a good use case. I personally don't see where we would need to filter a min/max GPA especially since each GPA is mapped to a course-instructor pairing. I guess the question to ask is, "What do we gain (whether it be better data analysis accuracy or new information) by filtering out GPAs that are lower/higher than a threshold we determine?". On the other hand, filtering out with min/max on a GPA tied to only a course would have more realistic applications, but this would require us to calculate the average GPA for a course as well.

I agree with removing 'student, withdraws, percentages of students of each grade, and GPA'. With the logic flow of the program, we really only need to query for classes, instructors, and course names for now.

Reply:
Makes sense. It wouldn't be costly to search all classes for a subject and then filter from there. Let's remove all of the things you said in  the last part.

Reply:
Done.